kratos/includes/kratos_application.h

```diff
@@ -211,49 +207,25 @@ class KRATOS_API(KRATOS_CORE) KratosApplication {
             VariableDataComponents)
 
     {
-        for (KratosComponents<VariableData>::ComponentsContainerType::iterator
-                 i = mpVariableData->begin();
-
-             i != mpVariableData->end(); i++)
-
-        {
-            std::string const& variable_name = i->second->Name();
-            KratosComponents<VariableData>::ComponentsContainerType::
-                const_iterator i_variable =
-                    VariableDataComponents.find(variable_name);
-
-            if (i_variable == VariableDataComponents.end())
+        for (auto it = mpVariableData->begin(); it != mpVariableData->end(); it++) {
+            std::string const& r_variable_name = it->second->Name();
+            auto it_variable = VariableDataComponents.find(r_variable_name);
 
-                KRATOS_THROW_ERROR(std::logic_error,
-                    "This variable is not registered in Kernel : ",
-                    *(i_variable->second));
+            KRATOS_ERROR_IF(it_variable == VariableDataComponents.end()) << "This variable is not registered in Kernel : " << *(i_variable->second)) << std::endl;
 
-            unsigned int variable_key = i_variable->second->Key();
+            const std::size_t variable_key = it_variable->second->Key();
 
-            if (variable_key == 0)
+            KRATOS_ERROR_IF(variable_key == 0) << This variable is not initialized in Kernel : " << *(i_variable->second)) << std::endl;
 
-                KRATOS_THROW_ERROR(std::logic_error,
-                    "This variable is not initialized in Kernel : ",
-                    *(i_variable->second));
-
-            //			KRATOS_WATCH(i_variable->second.get());
-
-            //			KRATOS_WATCH(i->second.get().Key());
-
-            //			KRATOS_WATCH(variable_key);
-
-            i->second->SetKey(variable_key);
+            it->second->SetKey(variable_key);
```

Comment:
ping @pooyan-dadvand does this even make sense any more? I thought now the keys are compile-time constants?

Reply:
Not anymore. It can be removed safely.